templates/shared/forms/_form-validation.html

```diff
@@ -0,0 +1,175 @@
+    var return_url = marketoForm.querySelector('input[name=return_url]');
+    if (return_url != null) {
+      return_url = return_url.value;
+    }
+
+    // check if it is a dynamic modal form
+    var isModal = marketoForm.classList.contains("modal-form");
+
+    // check if there is a thank you message to post
+    var thankYouMessage = marketoForm.querySelector('input[name=thankyoumessage]');
+    if (thankYouMessage != null) {
+      thankYouMessage = thankYouMessage.value;
+    }
+    formid = this.id;
+
+    console.log("download_asset_url: " + download_asset_url);
+    console.log("return_url: " + return_url);
+    console.log("modal: " + isModal);
+    console.log("thankYouMessage: " + thankYouMessage);
+
+    // deal with the post submit actions
+    afterSubmit(download_asset_url, return_url, isModal, thankYouMessage);
+
+    return true;
+  }
+
+  /**
+  * After submit has happened
+  * start download and send the user to the instructions page
+  */
+
+  afterSubmit = function(download_asset_url, return_url, isModal, thankYouMessage) {
+
+    // Now start the download
+    if (download_asset_url) {
+      var downloadFrame = document.createElement("iframe");
+      downloadFrame.setAttribute("src", download_asset_url);
+      downloadFrame.style.width = 0 + "px";
+      downloadFrame.style.height = 0 + "px";
+      document.body.insertBefore(downloadFrame, document.body.childNodes[0]);
+    }
+
+    // And redirect to the instructions page
+    if (return_url) {
+      window.setTimeout(function() {
+        window.location.href = return_url;
+      }, 1000);
+    }
+
+    // dynamic thank you HACK
+    if (isModal) {
+      document.getElementsByClassName('js-pagination--3')[0].classList.add("u-hide");
+      document.getElementsByClassName('js-pagination--4')[0].classList.remove("u-hide");
+    }
+
+    // add a thank-you notification to the page
+    // if someone submitted a form without a thank you action
+    var thankYouMessage = thankYouMessage;
+    if (return_url === null || isModal === false) {
+      console.log("thank you: " + thankYouMessage);
+      if (thankYouMessage === null) {
+        thankYouMessage = 'Thank you<br />A member of our team will be in touch within one working day';
+      }
+      document.getElementById('main-content').insertAdjacentHTML('afterbegin', '<div class="p-strip is-shallow u-no-padding--bottom"><div class="row"><div class="p-notification--positive"><p class="p-notification__response"><span class="p-notification__status">Success:</span> ' + thankYouMessage + '</p></div></div></div>');
+      window.scrollTo(0,0);
+    }
+  }
+
+  // recaptcha submitCallback
+  var CaptchaCallback = function() {
+    let recaptchas = document.querySelectorAll("div[class^=g-recaptcha]");
+    recaptchas.forEach(function(field){
+      recaptchaWidgetId = grecaptcha.render(field, {'sitekey' : '6LfYBloUAAAAAINm0KzbEv6TP0boLsTEzpdrB8if'});
+      field.setAttribute("data-widget-id", recaptchaWidgetId);
```

Comment:
You are using data-widgetId above. Should they be the same?

Reply:
no?